pkg/common/cluster_state.go

```diff
@@ -310,7 +310,7 @@ func (i *ClusterState) readDatabaseSecretCurrentState(context context.Context, c
 }
 
 func (i *ClusterState) readKeycloakOrRHSSODeploymentCurrentState(context context.Context, cr *kc.Keycloak, controllerClient client.Client) error {
-	isRHSSO := cr.Spec.Profile == "RHSSO"
+	isRHSSO := cr.Spec.Profile == RHSSOProfile
```

Comment:
Definitely not! I would like to have profile described very carefully in productized docs.

Reply:
Soudsn good, we need to add some CRD validation as well. This can be done in a separate PR if needed. 
`// +kubebuilder:validation:Enum=RHSSO` should be added above the field in the spec and `make code/gen` run again. We should really be adding other validation markers to the spec and status as well